src/dex/camelot/camelot.ts

```diff
@@ -205,8 +213,58 @@ export class Camelot
 
     this.routerInterface = new Interface(ParaSwapABI);
     this.exchangeRouterInterface = new Interface(UniswapV2ExchangeRouterABI);
+
+    this.factoryInst = new UniswapV2Factory(
+      dexHelper,
+      dexKey,
+      factoryAddress,
+      this.logger,
+      this.onPoolCreatedDeleteFromNonExistingSet,
+    );
+  }
+
+  private _constructPoolKey(token0: string, token1: string) {
+    const [_token0, _token1] =
+      token0.toLowerCase() < token1.toLowerCase()
+        ? [token0, token1]
+        : [token1, token0];
+
+    const poolKey = `${token0}-${token1}`.toLowerCase();
+
+    return poolKey;
   }
 
+  /*
+   * When a non existing pool is queried, it's blacklisted for an arbitrary long period in order to prevent issuing too many rpc calls
+   * Once the pool is created, it gets immediately flagged
+   */
+  onPoolCreatedDeleteFromNonExistingSet: OnPoolCreatedCallback = async ({
+    token0,
+    token1,
+  }) => {
+    const logPrefix = '[onPoolCreatedDeleteFromNonExistingSet]';
+
+    try {
+      const [_token0, _token1] =
+        token0.toLowerCase() < token1.toLowerCase()
+          ? [token0, token1]
+          : [token1, token0];
+
+      const poolKey = this._constructPoolKey(token0, token1);
+
+      this.newlyCreatedPoolKeys.add(poolKey);
+
+      // delete entry locally to let local instance discover the pool
+      delete this.pairs[poolKey];
+
+      this.logger.info(`${logPrefix} discovered new pool ${poolKey}`);
+    } catch (e) {
+      this.logger.error(
+        `${logPrefix} LOGIC ERROR on ack new pool (token0=${token0},token1=${token1})`,
```

Comment:
@mwamedacen why we are not logging actual error here?

Reply:
Thx I will